packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -2223,28 +2215,22 @@ export function attach(
     const id = fiberInstance.id;
     const isRoot = fiber.tag === HostRoot;
     if (isRoot) {
-      // Roots must be removed only after all children (pending and simulated) have been removed.
+      // Roots must be removed only after all children have been removed.
       // So we track it separately.
       pendingUnmountedRootID = id;
     } else if (!shouldFilterFiber(fiber)) {
       // To maintain child-first ordering,
       // we'll push it into one of these queues,
       // and later arrange them in the correct order.
-      if (isSimulated) {
-        pendingSimulatedUnmountedIDs.push(id);
-      } else {
-        pendingRealUnmountedIDs.push(id);
-      }
+      pendingRealUnmountedIDs.push(id);
     }
 
-    if (!fiber._debugNeedsRemount) {
```

Comment:
I couldn’t figure out what this check was needed for. It’s not covered by any tests nor comments in the original PR. Some of the HMR issues that the PR refers to don’t have repros so it might have just been throwing things against the wall to see what sticks.

But this doesn’t make sense to me because if a fiber is going to get remounted it always a new fiber instance anyway which is disconnected and gets a new id so the old one does need to be cleaned up.